### ISSUE TITLE
Certain images cause noise artifacts?

### ISSUE THREAD

**Issue by GlitchBob452** [numz/ComfyUI-SeedVR2_VideoUpscaler#64]
Not sure what the issue could be.

Input:
<img width="720" height="574" alt="Image" src="https://github.com/user-attachments/assets/f2483f9d-dba9-4e86-aedb-3765c07a1f50" />

Output with 7B Sharp FP16 , set to 2048px:
![Image](https://github.com/user-attachments/assets/1465042f-936f-44dc-891f-07e94cf9af48)

Output with 7B FP16 , set to 2048px:
![Image](https://github.com/user-attachments/assets/8b7e2047-d6ad-4c1b-898b-f965fe9e243c)

Output with 3B FP16 , set to 2048px:
![Image](https://github.com/user-attachments/assets/bf2c8a11-4950-4338-97ba-cd026fcf8911)

**Comment by adrientoupet** [comment]
Thank you for sharing this example - I'll look into it.

**Comment by IceClear** [comment]
> [@GlitchBob452](https://github.com/GlitchBob452) Hi～ May you send all the images to me？I will have a test when I am free. iceclearwjy at gmail.com

Such failure case can be possible since the images seem to be very different from the natural one and the model seems to try to add some details, though like artifacts eventually.

**Comment by adrientoupet** [comment]
Hi there,

@IceClear I don't think it's a problem of training dataset. I spent a good part of yesterday looking into this issue. The model upscale this image well up to a certain resolution and then start creating a lot of noise artifacts. I'm not sure yet if it's a limitation of the ConfyUI integration, our quantized ComfyUI model or the original model. Do you know if there is any limitation around the attention window? My assumption with SeedVR2 was that the attention window was adaptive and working well at any output resolution. 

I'll keep digging but if you have insights about the original implementation, I'll take them. Thank you

**Comment by adrientoupet** [comment]
Hi @IceClear  - I finally managed to try on a bigger machine to test the original model

Command:
 torchrun --nproc-per-node=1 projects/inference_seedvr2_7b.py --video_path video_in --output_dir video_out --seed 123 --res_h 1472 --res_w 1840 --sp_size 1

Source:
https://github.com/user-attachments/assets/3e282b40-b557-43cc-b590-a5952b62e84c
<img width="720" height="574" alt="Image" src="https://github.com/user-attachments/assets/7c5bc861-4d30-45aa-873b-b4731f35820d" />

Upres
https://github.com/user-attachments/assets/4547970f-c14f-4dba-8765-7286c6f4da11
<img width="1840" height="1456" alt="Image" src="https://github.com/user-attachments/assets/107088bf-6dfe-4fd0-9239-21a9e40c81aa" />

This makes me really sad - That means the issue is in the original model/code. As @GlitchBob452 mentioned, it upscale fine up to height of around 1300px then start adding a lot of undesirable noise in areas that are meant to be constant.
As mentioned earlier, I tested with other images and saw similar issue.

Here is an example of a HD picture of people:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/f041d5bc-9e98-434b-b406-4361ac320f49" />

Trying to upscale it to 2128x1200, this time in ComfyUI with the 7b_fp16 model and no other optimisation:

<img width="2128" height="1200" alt="Image" src="https://github.com/user-attachments/assets/745d5c67-56d6-4005-a49e-de623dc2808c" />

And pushing it to 2832x1600:

<img width="2832" height="1600" alt="Image" src="https://github.com/user-attachments/assets/71dd5dfb-1dce-42b5-9647-8a66f74e95a2" />

 Can you please do a quick test on your side, confirm the issue and let me know where we go from there? I'd love to help resolve this but I'm not sure where to start if the issue is in the original model/implementation.

Thank you for your help

**Comment by IceClear** [comment]
Hi~ @adrientoupet, thanks for your feedback! 
Unfortunately, the maximum resolution I can handle on my original code is around 2K, and I cannot handle a 2832x1600 image. While it may not be proper for me to install ComfyUI on internal servers, I have some hints fo your reference:

1. From my side, it seems like there is a resolution upper bound for the current attention mechanism, unfortunately. You may try to adjust the upper bound of the window size [here](https://github.com/ByteDance-Seed/SeedVR/blob/main/models/dit/window.py#L57) to see if a larger window size can enable good performance on 2832x1600.
2. If this does not work, we may have to go back to the traditional patch-wise operation. We need to divide the large inputs into relatively small patches, as previous methods, e.g., Upscale-A-Video and STAR. But our advantage here is that we can use flexible patch sizes, i.e, 1k patches to make this process much more efficient. You may refer to this figure from StableSR:

<img width="753" height="482" alt="Image" src="https://github.com/user-attachments/assets/f01129cd-6b88-47af-a760-83c4173227cf" />

The reason behind this, I conjecture, is related to the distribution gap between training and inference. Though the window attention can alleviate this gap, given the same resolution of a patch, the distribution across different resolutions can vary a lot.
Feel free to reach me out if you have other concerns or want to know more details about how to solve this from the code level.

**Comment by adrientoupet** [comment]
Thank you @IceClear for your feedback. 

I tried a lot of things with the window size, even removing windowing entirely, but that didn't solve the artifact.
Ideally I'd like not to have to go back to traditional patch-wise operation. You did a lot of work to avoid this, would be nice to not change the architecture now.

However I found that increasing the cond_noise_scale in the generation.py when doing high resolution upscaling allows to get rid of the artifcats.

This is the same image that was failing earlier,  now upsscaled without artifacts at 1840x1472:

<img width="1840" height="1472" alt="Image" src="https://github.com/user-attachments/assets/00ddeb93-ab6d-4649-b9bc-8349ed6be82f" />

I don't want to introduce cond noise when output latents are lower than 90x90, because it's already working well with cond noise 0. Im still refining the magic curve to introduce cond noise scale to work at any output resolution above 90x90 latent. But will likely need to expose also that parameter to the user so they can dial that number if needed. I don't want to increase it too much by default because it suppresses details and smooth out everything.

Before I pursue with this approach, can you confirm if this is the right way to go? Or if you have any concern? Thank you

**Comment by adrientoupet** [comment]
All those annoying artifacts bugs should now be resolved in the latest release. 
Please update, to the latest version of SeedVR2 in ComfyUI Manager, watch the tutorial to see how it all fits together - https://www.youtube.com/watch?v=MBtWYXq_r60 - and reopen this thread if you're still running into issue.
